test/unit/SES/Transport/SESTransportTest.php

```diff
@@ -0,0 +1,228 @@
+    {
+        return $this->dummy_mime_string;
+    }
+}
+
+class SESTransportTest extends TestCase
+{
+    protected Swift_Events_EventDispatcher $event_dispatcher;
+
+    protected SpyingSESClient $ses_client;
+
+    protected ArrayMetricsAgent $metrics;
+
+    protected TestLogger $log;
+
+    public function test_it_is_initialisable()
+    {
+        $this->assertInstanceOf(SESTransport::class, $this->newSubject());
+    }
+
+    public function test_it_is_started()
+    {
+        $this->assertTrue($this->newSubject()->isStarted());
+    }
+
+    public function test_it_throws_with_more_than_one_from_address()
+    {
+        $message = new Swift_Message;
+        $message->addFrom('foo@bar.com');
+        $message->addFrom('other@bar.com');
+        $subject = $this->newSubject();
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('multiple `from` address');
+        $subject->send($message);
+    }
+
+    public function test_it_throws_with_more_than_one_to_address()
+    {
+        $message = new Swift_Message;
+        $message->addTo('foo@bar.com');
+        $message->addTo('other@bar.com');
+        $subject = $this->newSubject();
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('multiple `to` address');
+        $subject->send($message);
+    }
+
+    public function test_its_register_plugin_just_binds_to_event_listener()
+    {
+        $plugin  = new DummySwiftPlugin;
+        $subject = $this->newSubject();
+        $subject->registerPlugin($plugin);
+
+        $evt = $this->event_dispatcher->createSendEvent($subject, $this->getDummy(Swift_Mime_SimpleMessage::class));
+        $this->event_dispatcher->dispatchEvent($evt, 'sendPerformed');
+        $plugin->assertGotEvents([['sendPerformed', $evt]]);
+    }
+
+    public function test_its_send_method_sends()
+    {
+        $subject = $this->newSubject();
+        $message = $this->generateValidMessage();
+        $subject->send($message);
+        $this->assertSame([['RawMessage' => ['Data' => $message->toString()]]], $this->ses_client->getSentMails());
+    }
+
+    public function test_its_start_is_no_op()
+    {
+        $this->assertTrue($this->newSubject()->start());
+    }
+
+    public function test_its_stop_is_no_op()
+    {
+        $this->assertTrue($this->newSubject()->stop());
+    }
+
+    public function test_it_emits_timer_metric()
+    {
+        $subject = $this->newSubject();
+        $subject->send($this->generateValidMessage());
+        $this->metrics->assertCapturedOneTimer('ses-email-sent', 'ok');
+    }
+
+    public function test_it_emits_timer_metric_with_err_src_on_failure()
+    {
+        $this->ses_client = SpyingSESClient::sendRawEmailWillThrow();
+        $subject          = $this->newSubject();
+        try {
+            $subject->send($this->generateValidMessage());
+        } catch (\Exception $e) {
+        }
```

Comment:
worth asserting something about the specific thrown exception here (instance / message / class)? Just so if something else breaks in the future it's more obvious for debugging, rather than looking like it's just that the metric never got captured?